src/utils/type-fu.ts

```diff
@@ -0,0 +1,7 @@
+export type Keys<T> = keyof T
+
+export type Values<T> = T[keyof T]
+
+export type Equals<X, Y> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y ? 1 : 2 ? true : false
```

Comment:
This is doing a lot of heavy lifting in this PR to ensure that our zod schemas match the `roo-code.d.ts` type definitions. Ideally we'd infer the types from the schema and nicely DRY things up, but the type definition file can only have type declarations in it.